sdk/turing/router/router.py

```diff
@@ -130,20 +130,17 @@ def get(cls, router_id: int) -> 'Router':
         return Router.from_open_api(
             turing.active_session.get_router(router_id=router_id))
 
-    def update(self, config: RouterConfig) -> 'Router':
+    def update(self, config: RouterConfig) -> RouterVersion:
         """
         Update the current router with a new set of configs specified in the RouterConfig argument
 
         :param config: configuration of router
-        :return: instance of router (self); this router contains details of the currently deployed router; it contains
-            the details of the currently DEPLOYED router version
+        :return: the new router version
         """
-        self._config = config
-        updated_router = Router.from_open_api(
-            turing.active_session.update_router(router_id=self.id, router_config=config.to_open_api())
-        )
-        self.__dict__ = updated_router.__dict__
-        return self
+        version = turing.active_session.update_router(router_id=self.id,
+                                                      router_config=config.to_open_api(),
+                                                      deploy=True)
+        return RouterVersion(environment_name=self.environment_name, name=self.name, **version.to_dict())
```

Comment:
The SDK now reflects the change in the response when updating a router (with deployment). It now expects a `RouterVersion` instead of a updated `Router`.